### PR TITLE
show cilium_vxlan as decryption interface in tunnel mode

### LINKED ISSUE #40373: Empty interfaces for tunnel mode in `cilium encrypt status`
When in tunnel mode, the output is a bit confusing:
```
Encryption: IPsec
Decryption interface(s):
Keys in use: 16
Max Seq. Number: 0x329e/0xffffffff
Errors: 0
```
This is probably just because we only display native interfaces and we're not attaching to any native interfaces in tunnel mode. We only attach to cilium_vxlan.

We should check that we are indeed attached to cilium_vxlan's ingress and display that instead of nothing. Displaying nothing leads some users to believe there's an issue.

### PR DESCRIPTION
**Pull request by AritraDey-Dev**
Fixes #40373
show cilium_vxlan as decryption interface in tunnel mode
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Fixes: #40373

```release-note
cilium-dbg encrypt status now displays the tunnel device (e.g. cilium_vxlan) when running in tunnel mode without native interfaces attached
```

### CODE REVIEW

**Comment by joestringer** [comment]
@ldelossa I see the `dont-merge/discussion` label is assigned to this PR. Is that discussion ongoing? Or is it fair to say that you think that this problem needs more thought around the user experience for the command before reviewing a PR like this one?

File: cilium-dbg/cmd/encrypt_status.go [hunk 1]
```diff
@@ -92,6 +93,9 @@ func dumpIPsecStatus() (models.EncryptionStatus, error) {
 	if err != nil {
 		return models.EncryptionStatus{}, fmt.Errorf("error getting IPsec decryption interfaces: %w", err)
 	}
+	if option.Config.TunnelingEnabled() && len(decryptInts) == 0 {
+		decryptInts = append(decryptInts, "cilium_vxlan")
```
Comment (joestringer) [inline cilium-dbg/cmd/encrypt_status.go]: This conditional checks `TunnelingEnabled` but then prints the string `cilium_vxlan`. Not all tunneling modes are vxlan. The device should correspond to the tunnel device type in use.

**Comment by pchaigno** [comment]
> @ldelossa I see the `dont-merge/discussion` label is assigned to this PR. Is that discussion ongoing? Or is it fair to say that you think that this problem needs more thought around the user experience for the command before reviewing a PR like this one?

@joestringer There's a discussion at https://github.com/cilium/cilium/issues/40373#issuecomment-3045683864.

**Comment by ldelossa** [comment]
Removing the discussion label blocker. Since @pchaigno outlined a solution which appeals to both of us.

File: cilium-dbg/cmd/encrypt_status.go [hunk 2]
```diff
@@ -92,6 +93,9 @@ func dumpIPsecStatus() (models.EncryptionStatus, error) {
 	if err != nil {
 		return models.EncryptionStatus{}, fmt.Errorf("error getting IPsec decryption interfaces: %w", err)
 	}
+	if option.Config.TunnelingEnabled() && len(decryptInts) == 0 {
```
Comment (ldelossa) [inline cilium-dbg/cmd/encrypt_status.go]: @pchaigno correct me if I'm wrong, but in the original issue, I think the suggestion was to actually find where we attach our eBPF program to that does the decap processing? This solution looks a bit too simple.

Comment (pchaigno) [reply cilium-dbg/cmd/encrypt_status.go]: Yes, I think it would be good to check that the BPF program is attached to the overlay device, as we do for the native devices. And that logic should be folded into `getDecryptionInterfaces`.

**Review by pchaigno — CHANGES_REQUESTED** [review]
Requesting changes based on above discussion with Louis.

File: cilium-dbg/cmd/encrypt_status.go [hunk 3]
```diff
@@ -185,40 +185,176 @@ func maxSequenceNumber() (string, error) {
...
+	// Map tunnel protocol to device name
+	switch tunnelProtocol {
+	case "vxlan":
+		return "cilium_vxlan", nil
+	case "geneve":
+		return "cilium_geneve", nil
+	default:
+		return "", fmt.Errorf("unsupported tunnel protocol: %v", tunnelProtocol)
+	}
+}
```
Comment (pchaigno) [inline cilium-dbg/cmd/encrypt_status.go:395]: I see [CI fail](https://github.com/cilium/cilium/actions/runs/17727775368/job/50372199602) with:
```
failed to fetch encryption status from cilium-5zn7v: command failed (pod=kube-system/cilium-5zn7v, container=cilium-agent): command terminated with exit code 1: "Error: Cannot get daemon encryption status: error getting IPsec decryption interfaces: failed to check tunnel decryption interface: failed to get tunnel device cilium_vxlan: Link not found\n"
```
so I suspect there's an issue with this function.

**Review by pchaigno — CHANGES_REQUESTED** [review]
Thanks for the update!

Could you add more context to the commit description (please wrap lines at 75 characters) and describe how you tested this?

**Comment by AritraDey-Dev** [comment]
> Could you add more context to the commit description (please wrap lines at 75 characters) and describe how you tested this?

tested this by building the cilium-dbg binary,in cilium cluster and running `encrypt status` with and without tunnel devices.
Also added unit tests to cover the changes. 
Should we consider adding more unit tests for edge cases, such as empty program names,unknown program names,partial matches?

**Comment by pchaigno** [comment]
> tested this by building the cilium-dbg binary,in cilium cluster and running `encrypt status` with and without tunnel devices.
> Also added unit tests to cover the changes.

Nice, thanks!

> Should we consider adding more unit tests for edge cases, such as empty program names,unknown program names,partial matches?

I think it's okay as is.

The commit description still needs more context. You can take some from the issue you're fixing.

**Comment by joestringer** [comment]
Could you also update the release note? This doesn't seem to properly describe what is changing:

> Tunnel mode now shows cilium_vxlan interface.

**Comment by joestringer** [comment]
@cilium/cli Looks like this is still waiting on your review

**Review by mhofstetter — APPROVED** [review]
looks good from a pure CLI perspective. i thought there are still ongoing "encryption" related discussions - and was expecting a approval from one of the encryption reviewers. sorry for the delay! maybe worth to change codeowners for these files anyway.

Note: this needs a rebase.

**Comment by pchaigno** [comment]
As noticed by Bernardo in https://github.com/cilium/cilium/issues/40373#issuecomment-3557116294, this shouldn't be necessary since we now have Encrypted Overlay.

**Comment by aanm** [comment]
@AritraDey-Dev are you still working on this PR? If yes, feel free to mark it "ready for review"
